modules/ROOT/pages/dataweave-selecting-types.adoc

```diff
@@ -0,0 +1,204 @@
+
+== Declare a New Type from an Existing Type
+The following example declares the `NameType` from the `User` type. Using the dot selector in `User.name` to declare the `name` variable a `NameType` enables the `typeOf()`  function to return the basic type, `"String"`. Notice how the DataWeave variable `name` is declared as a `NameType` value "Seba":
+
+.DataWeave script:
+[source,dataweave,linenums]
+----
+%dw 2.0
+type User = {name: String}
+type NameType = User.name
+var name: NameType = "Seba"
+---
+typeOf(name)
+----
+
+.Output: 
+----
+"String"
+----
+
+== Compose a Type from an Existing Type's Field
+The following example uses the dot selector (`.`) to declare a new type `Address` from one key in the complex type `User`:
+
+.DataWeave script:
+[source,dataweave,linenums]
+----
+%dw 2.0
+output application/json
+type User = {
+ address : {country: String, city: String, street: String, number: Number},
+ userName : String
+}
+type Address = User.address
+
+var userAddress: Address = {country: "Argentina", city: "Rosario", street: "Calle Falsa", number: 123}
+---
+userAddress
+----
+
+.Output:
+[source,json,linenums]
+----
+{
+ "country": "Argentina",
+ "city": "Rosario",
+ "street": "Calle Falsa",
+ "number": 123
+}
+----
+
+== Select Types from Union Types
+The following example shows how type selection can also be used with `Union` types that consist of `Object` types:
+
+.DataWeave script:
+[source,dataweave,linenums]
+----
+%dw 2.0
+output application/json
+type ID = {name: String} | {id: Number}
+var name: ID = {name: "Seba"}
+var id: ID = {id: 38123}
+var nameString: ID.name = "Seba"
+var idNumber: ID.id = 38123
+---
+[name, id, nameString ++ " is of type " ++ typeOf(nameString), idNumber ++ " is of type " ++ typeOf(idNumber)]
+----
+
+.Output:
+[source,json,linenums]
+----
+[
+  {
+    "name": "Seba"
+  },
+  {
+    "id": 38123
+  },
+  "Seba is of type String",
+  "38123 is of type Number"
+]
+----
+
+However, the following script fails because the second type is not an `Object` type and there is no way to reference it with the selector:
+
+.DataWeave script:
+[source,dataweave,linenums]
+----
+%dw 2.0
+output application/json
+type UnionType = {name: String} | Number
+type Fail = UnionType.name
+---
+{}
+----
+
+.Output:
+[source,json,linenums]
+----
+Cannot do selection on Type: Number
+
+4| type Fail = UnionType.name
+----
+
+== Select Types from Namespaces
+The following example shows type selection of a namespace:
+
+.DataWeave script:
+[source,dataweave,linenums]
+----
+%dw 2.0
+output application/json
+ns ns1 http://acme.com
+type User = {
+ ns1#name : String,
+ name : Number,
+}
+var nameString: User.ns1#name = "Seba"
+var nameNumber: User.name = 123
+---
+{
+   "NameString" : typeOf(nameString),
+   "NameNumber" : typeOf(nameNumber)
+}
+----
+
+.Output:
+[source,json,linenums]
+----
+{
+ "NameString": "String",
+ "NameNumber": "Number"
+}
+----
+
+== Select Types to Preserve Metadata
```

Comment:
This title is kind of hard to construct. Maybe I'm overthinking it, but rather than selecting _to_ (with the goal of?) preserving metadata, the example is showing that the metadata of the type is preserved when you select it. So, for ex, if the variable `otherUserName` used `{schema: "value"}` vs. `{schema: "otherValue"}` in its definition, then it would return `true` instead of `false`. Basically, it's letting us know that you don't lose the metadata when you use the selector. I'm not sure how to convey that in the title, tbh.

Reply:
Decided on "Preserve Metadata of a Type".